gobblin-runtime/src/main/java/org/apache/gobblin/runtime/TaskStateCollectorService.java

```diff
@@ -190,13 +190,13 @@ protected void shutDown() throws Exception {
    */
   private void collectOutputTaskStates() throws IOException {
 
-    final Queue<TaskState> taskStateQueue = deserializeTaskStatesFromFolder(taskStateStore, outputTaskStateDir, this.stateSerDeRunnerThreads);
-    if (taskStateQueue == null) {
+    final Optional<Queue<TaskState>> taskStateQueue = deserializeTaskStatesFromFolder(taskStateStore, outputTaskStateDir.getName(), this.stateSerDeRunnerThreads);
+    if (!taskStateQueue.isPresent()) {
       return;
     }
     // Add the TaskStates of completed tasks to the JobState so when the control
     // returns to the launcher, it sees the TaskStates of all completed tasks.
-    for (TaskState taskState : taskStateQueue) {
+    for (TaskState taskState : taskStateQueue.get()) {
```

Comment:
nit: the farther down we get, the harder to keep track of whether these `.get()` calls are truly safe (i.e. previously checked).  the common convention to establish safety looks like:
```
if (!optTSQ.isPresent()) {
  return;
} // could be `else`, if you don't mind indentation

Queue<TaskState> tsq = optTSQ.get();
// (henceforth, no more mention of `optTSQ`)
...
```